### ISSUE TITLE
filter dovecot.conf needs another line to work properly for me

### ISSUE THREAD

**Issue by treveradams** [fail2ban/fail2ban#1897]
Currently there is for dovecot a line 
```
            ^pam\(\S+,<HOST>\): pam_authenticate\(\) failed: (User not known to the underlying authentication module: \d+ Time\(s\)|Authentication failure \(password mismatch\?\))\s*$
```
This does not catch all things (for me nothing), I have to have 
```
            ^pam\(\S+,<HOST>,\S+\): pam_authenticate\(\) failed: Permission denied\s*$
```
as well.

per https://bugzilla.redhat.com/show_bug.cgi?id=1356573

**Comment by sebres** [comment]
Please tell us here an example (the log-excerpt) contains this kind of failures...

**Comment by treveradams** [comment]
Sep 22 11:39:13 EMAIL_SERVER_FQDN dovecot[24941]: auth-worker(30165): pam(user@DOMAIN,127.0.0.1,<PNHQq8pZhqIKAQGd>): pam_authenticate() failed: Permission denied

This has obviously been edited to remove what might be considered sensitive information, but it is otherwise an actual log entry.

fail2ban-0.10.0-1.fc27.noarch
fail2ban-firewalld-0.10.0-1.fc27.noarch
fail2ban-sendmail-0.10.0-1.fc27.noarch
fail2ban-server-0.10.0-1.fc27.noarch
fail2ban-systemd-0.10.0-1.fc27.noarch

dovecot-2.2.32-2.fc27.x86_64
dovecot-pigeonhole-2.2.32-2.fc27.x86_64

**Comment by sebres** [comment]
It looks like dovecot configuration issue (I guess dovecot auth-worker cannot use `pam_authenticate` rather as real authentication failure)...

**Comment by treveradams** [comment]
No, it is a real authentication error. The setup works with the correct user/password. The error in question is when I deliberately entered an incorrect password. My setup is configured for kerberos (gssapi, or plain) and pam (via pam_krb5). It is possible the difference is because of a kerberos return, I do not know.

**Comment by sebres** [comment]
> The setup works with the correct user/password. 

This alone says nothing.
Using multiple authenticator

**Comment by treveradams** [comment]
No. The account being tested cannot/does not use Kerberos due to various reasons (client related). The error is due to authentication failure and none of the rules currently in the regex match.

My Thunderbird mail client has three accounts configured. One Kerberos, two others in two separate (but similarly configured) domains. The other two accounts are always plain text password over SSL. The password is checked via pam_krb5, no other method. The log entry in question only shows when user name and/or password (although I must say I haven't checked incorrect user name in a long time) is incorrect.

As I said, it works with correct password, no log entry.
With incorrect password, account is not accessible and the log entry exists.
The account in question is NEVER authenticated nor attempted to be accessed with Kerberos.

**Comment by treveradams** [comment]
I understand you are saying it is possible to get it for another reason, but it also means authentication failure due to bad password with pam_krb5.

**Comment by treveradams** [comment]
I am seeing that if the KDC is down, the error can happen as well.

**Comment by sebres** [comment]
I think it will not bother to accept dovecot pam_authenticate failures with "Permission denied" message as authentication failure...
Thus fixed in ea36e1b3fc22f67b2faedb8cb57ae1b19d9f6e08 and merged in all major branches.
